zip/zip.go

```diff
@@ -0,0 +1,95 @@
+package zip
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/dsnet/compress/bzip2"
+)
+
+func CreateArchiveFromDir(source string) error {
+	tarFile, err := os.Create(source + ".tar.bz2")
+	if err != nil {
+		return err
+	}
+	defer tarFile.Close()
+
+	bz2Writer, err := bzip2.NewWriter(tarFile, &bzip2.WriterConfig{Level: bzip2.BestCompression})
+	if err != nil {
+		return err
+	}
+	defer bz2Writer.Close()
+
+	tarWriter := tar.NewWriter(bz2Writer)
+	defer tarWriter.Close()
+
+	err = filepath.Walk(source, func(file string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		header, err := tar.FileInfoHeader(fi, file)
+		if err != nil {
+			return err
+		}
+
+		header.Name = filepath.ToSlash(file)
+		if err := tarWriter.WriteHeader(header); err != nil {
+			return err
+		}
+		if !fi.Mode().IsRegular() {
+			return nil
+		}
+
+		fileReader, err := os.Open(file)
+		if err != nil {
+			return err
+		}
+		defer fileReader.Close()
+
+		_, err = io.Copy(tarWriter, fileReader)
```

Comment:
scoping

Reply:
In this case, I return the `err` at the end, so this would hurt readability.